animations/lib/src/misc/curved_animation.dart

```diff
@@ -0,0 +1,146 @@
+// Copyright 2019 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'dart:math' as math;
+
+class CurvedAnimationDemo extends StatefulWidget {
+  static const String routeName = '/misc/curved_animation';
+  @override
+  _CurvedAnimationDemoState createState() => _CurvedAnimationDemoState();
+}
+
+class MyCurve {
+  Curve curve;
+  String curveName;
+  MyCurve({this.curve, this.curveName});
+}
+
+class _CurvedAnimationDemoState extends State<CurvedAnimationDemo>
+    with SingleTickerProviderStateMixin {
+  static const String routeName = '/misc/curved_animation';
+  AnimationController animController;
+  Animation<double> animation;
+  static const _duration = Duration(seconds: 4);
+  TextStyle style = TextStyle(fontSize: 18.0);
+  List<MyCurve> forwardCurves = [
+    MyCurve(curve: Curves.bounceIn, curveName: 'Bounce In'),
+    MyCurve(curve: Curves.easeInCubic, curveName: 'Ease In Cubic'),
+    MyCurve(curve: Curves.easeInExpo, curveName: 'Ease In Expo'),
+    MyCurve(curve: Curves.elasticIn, curveName: 'Elastic In'),
+    MyCurve(curve: Curves.easeInQuart, curveName: 'Ease In Quart'),
+    MyCurve(curve: Curves.easeInCirc, curveName: 'Ease In Circ'),
+  ];
+  List<MyCurve> reverseCurves = [
+    MyCurve(curve: Curves.bounceOut, curveName: 'Bounce Out'),
+    MyCurve(curve: Curves.easeOutCubic, curveName: 'Ease Out Cubic'),
+    MyCurve(curve: Curves.easeOutExpo, curveName: 'Ease Out Expo'),
+    MyCurve(curve: Curves.elasticOut, curveName: 'Elastic Out'),
+    MyCurve(curve: Curves.easeOutQuart, curveName: 'Ease Out Quart'),
+    MyCurve(curve: Curves.easeOutCirc, curveName: 'Ease Out Circ')
+  ];
+  MyCurve selectedForwardCurve, selectedReverseCurve;
+  CurvedAnimation curvedAnimation;
+  @override
```

Comment:
nit: Add a line of whitespace here to separate the curvedAnimation property from the initState method. As you can see in the other code in the repo, it's typical to see a line of whitespace between fields of any kind.